scripts/theo/android/generate-values.ts

```diff
@@ -0,0 +1,96 @@
+import { writeFileSync } from 'fs';
+import { join } from 'path';
+import {
+  convert,
+  Format,
+  Prop,
+  registerFormat,
+  registerTransform,
+  registerValueTransform,
+} from 'theo';
+import xml from 'xml';
+import { opacity } from '../helpers';
+
+const valuesName = process.env.VALUES_NAME;
+if (!valuesName) throw new Error('No values name provided');
+
+const fileSuffix = process.env.NIGHT_MODE ? '-night' : '';
+
+const srcPath = join(__dirname, `src/values/${valuesName}${fileSuffix}.json`);
+const destPath = join(
+  __dirname,
+  `libraries/android/castor-tokens/src/main/res/values${fileSuffix}/castor-${valuesName}.xml`
+);
+
+registerValueTransform(
+  'ods/color/rgba',
+  (prop: Prop) => prop.get('type') === 'color',
+  (prop: Prop) => `rgba(${prop.get('value')}, ${opacity(prop)})`
+);
+
+registerTransform('android', [
+  'ods/color/rgba',
+  // keep default Theo transforms
+  'color/hex8rgba',
+  'relative/pixelValue',
+  'absolute/dp',
+  'percentage/float',
+]);
+
+(
+  [
+    {
+      type: 'android-color.xml',
+      propTypeFilter: 'color',
+      resourceKey: 'color',
+    },
+    {
+      type: 'android-dimen.xml',
+      propTypeFilter: 'size',
+      resourceKey: 'dimen',
+    },
+  ] as const
+).forEach(({ type, propTypeFilter, resourceKey }) => {
+  registerFormat(type, (result) => {
+    const obj = {
+      resources: result
+        .get('props')
+        .map((prop: Prop) => {
+          if (prop.get('type') !== propTypeFilter) return;
```

Comment:
Would a filter work or is there some type shenanigans?

```suggestion
        .filter((prop: Prop) => prop.get('type') === propTypeFilter)
        .map((prop: Prop) => {
```

Reply:
Yes, it should work just fine.